### DISCUSSION TITLE
Documention of mcsim

### DISCUSSION THREAD

**Discussion by wilberforce** [Moddable-OpenSource/moddable#855] [Getting started]
I have worked out how to launch:

`iot-product-dev-book\ch10-piu\host>mcconfig -d -m -p sim/moddable_two`

It appears you can then load new mods:
`iot-product-dev-book\ch10-piu\wifi-status>mcrun -d -m -p sim/moddable_two`

It does not seem to be part of the main documentation and I see there is a new M5paper target PR recently. I'm considering doing a M5StickC target.

**Comment by phoddie** [comment]
In terms of building and launching, `mcsim` is the same as the classic Screen Test simulator. It is not the default for builds at the moment, so you need to use `-p sim` or `-p simulator` to build, as you noted. When we change the default from Screen Test to `mcsim` that will not be necessary, unless you want to specify the device target (e.g. `-p simulator/modddable_two`). 

One of the reasons we are moving to `mcsim` is because it supports much more powerful device simulators. Where Screen Test is mostly about a device skin, `mcsim` makes it possible to simulate much more of the device. The details of that are currently undocumented. It is a big topic! If you are familiar with Piu and follow the examples, it is pretty straightforward to do though, as @FWeinb  shows with his excellent M5Paper simulator.

> I'm considering doing a M5StickC target

Cool!

**Reply by wilberforce** [reply]
M5 stick C simulator:
https://github.com/Moddable-OpenSource/moddable/pull/856

**Reply by phoddie** [reply]
(Delayed response!) This looks good. We'll pick it up in the next Moddable SDK push, probably tomorrow (Friday in California).
